### PR TITLE
Water mask documentation page

### PR DESCRIPTION
**Pull request by hjkristenson**
This PR adds a documentation page to the SAR Basics section describing the new water mask being used for InSAR processing. It also includes references to the subpackage in the asf_tools Python package for generating this mask, and lists the public bucket where the reference mask is stored.

### CODE REVIEW

File: docs/water_masking.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Water Mask
```
Comment (jtherrmann) [inline docs/water_masking.md:1]: Do we want to change this to `Water Masking` to match the name of the page (`water_masking`), or vice-versa?

File: docs/water_masking.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+The water mask rasters generated from the OSM and WorldCover datasets were mosaicked together, then tiled to 5° latitude by 5° longitude for storage. Because source data is only available from 85°S to 85°N, tiles were added to fill the polar areas. All pixels north of 85°N are treated as water, and all pixels south of 85°S are treated as land.
+
+This reference dataset is stored in a public AWS S3 bucket: 
+
+`s3://asf-dem-west/WATER_MASK/TILES/`
+
+In the reference raster dataset hosted in AWS, pixels with surface water are assigned a value of 1, and all other pixels are assigned a value of 0. 
+
+***Note that the pixel values used in the reference water mask are opposite to the pixel values used for the water masks included in the InSAR product packages. Refer to the [Applying the Water Mask section](#applying-the-water-mask-during-insar-processing "Jump to Applying the Water Mask during InSAR Processing section") for more information on how the reference water mask is transformed during InSAR processing.***
```
Comment (jtherrmann) [inline docs/water_masking.md:56]: I forgot what the triple-asterisk syntax (`***`) does, but I assume it renders as you'd expect?

File: docs/water_masking.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+If the option to [Apply Water Mask](https://hyp3-docs.asf.alaska.edu/guides/insar_product_guide/#apply-water-mask "InSAR Product Guide - Processing Options - Apply Water Mask" ){target=_blank} was selected by the user submitting the InSAR job, this mask is then used as an input, along with coherence values, to generate the [validity mask](https://hyp3-docs.asf.alaska.edu/guides/insar_product_guide/#masking "InSAR Product Guide - Masking" ){target=_blank} used for phase unwrapping. The 0-value water pixels are excluded from use in phase unwrapping.
+
+A copy of the water mask is always included in the InSAR product package for reference, even if the user chose not to select the option to apply the water mask. In this copy of the water mask, the pixel values are the same as what is used in InSAR processing: pixels indicating water have a value of 0, and all other pixels are assigned a value of 1.
```
Comment (jtherrmann) [inline docs/water_masking.md:74]: So why do we store the reference water mask with the opposite pixel values? I may have missed the explanation during one of our sprint reviews.

**Review by jtherrmann — APPROVED** [review]
LGTM, apart from my nitpicks / questions, which you're free to ignore if they're not relevant.

File: docs/water_masking.md (hunk 1 above)
Comment (forrestfwilliams) [reply docs/water_masking.md:1]: Yeah, I'd agree. Let's change it to "Water Masking".

File: docs/water_masking.md (hunk 3 above)
Comment (forrestfwilliams) [reply docs/water_masking.md:74]: It's a convention thing. Most InSAR tools expect that land is 1 and water is 0.

**Review by forrestfwilliams — DISMISSED** [review]
Approved, but as @jtherrmann noted I think we should change the title to "Water Masking"

File: docs/water_masking.md (hunk 3 above)
Comment (hjkristenson) [reply docs/water_masking.md:74]: @AndrewPlayer3 can give you more background, but I believe he didn't realize the mask was applied using the 0 values for water (in the hyp3-GAMMA process, the 0 values are applied to the validity mask; pixels with a 0 value for the mask are not included in phase unwrapping). When making the reference mask, true was 1 by default, so he just went with that, not realizing that it was the opposite convention from what's used in InSAR processing.

**Review by forrestfwilliams — APPROVED** [review]

**Review by jtherrmann — APPROVED** [review]
Re-re-approved :)

**Pull request merged**